metricbeat/module/haproxy/haproxy.go

```diff
@@ -243,3 +220,84 @@ func (c *Client) GetInfo() (*Info, error) {
 
 	return nil, err
 }
+
+type unixProto struct {
+	URL *url.URL
+}
+
+// Run sends a designated command to the haproxy stats socket
+func (p *unixProto) run(cmd string) (*bytes.Buffer, error) {
+	var conn net.Conn
+	response := bytes.NewBuffer(nil)
+
+	conn, err := net.Dial(p.URL.Scheme, p.URL.Host)
+	if err != nil {
+		return response, err
+	}
+	defer conn.Close()
+
+	_, err = conn.Write([]byte(cmd + "\n"))
+	if err != nil {
+		return response, err
+	}
+
+	_, err = io.Copy(response, conn)
+	if err != nil {
+		return response, err
+	}
+
+	if strings.HasPrefix(response.String(), "Unknown command") {
+		return response, fmt.Errorf("unknown command: %s", cmd)
+	}
+
+	return response, nil
+}
+
+func (p *unixProto) Stat() (*bytes.Buffer, error) {
+	return p.run("show stat")
+}
+
+func (p *unixProto) Info() (*bytes.Buffer, error) {
+	return p.run("show info")
+}
+
+type httpProto struct {
+	URL *url.URL
+}
+
+func (p *httpProto) Stat() (*bytes.Buffer, error) {
+	url := p.URL.String()
+	// Force csv format
+	if !strings.HasSuffix(url, ";csv") {
```

Comment:
What other formats do we have available?

Reply:
Only human-oriented html, so csv is the only reasonable one to parse :)

Reply:
JSON has been added to haproxy 1.8, but we may continue using CSV for backwards compatibility.

Reply:
Sounds like something we should probably add more details about to the docs.